caddyconfig/httpcaddyfile/pkiapp.go

```diff
@@ -16,23 +16,106 @@ package httpcaddyfile
 
 import (
 	"github.com/caddyserver/caddy/v2/caddyconfig"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
 	"github.com/caddyserver/caddy/v2/modules/caddypki"
 )
 
+func init() {
+	RegisterGlobalOption("pki", parsePKIApp)
+}
+
+// parsePKIApp parses the global log option. Syntax:
+//
+//     pki [id] {
```

Comment:
```suggestion
//     pki [<id>] {
```

To distinguish from a literal optional string "id".

Reply:
Also, I'm wondering if we can do a little better with the structure of this option. Right now all that can be done is configure CAs (which is all that can be done in the JSON too), but in the JSON we can add more without breaking existing configs, whereas here I don't know how we'd expand this to support more configuration later.

Reply:
How about like:
```
pki {
	some_imaginary_options
}
pki_ca [<id>] {

}
```

Reply:
Hmm, what about

```
pki {
    ca [<id>] {
       ...
    }
}
```

Reply:
I guess, just seems annoying to nest it when there's no other options yet :stuck_out_tongue: but sure why not

Reply:
Okay, done